### DISCUSSION TITLE
Code doesn´t works well<title>

### DISCUSSION THREAD

**Discussion by eduardodomit85** [community/community#43335] [Programming Help] [answered]
### Select Topic Area

Product Feedback

### Body

I´ve installed the application with this code: 
2023-01-04 18:36:37.501820: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2023-01-04 18:36:37.507891: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
WARNING:tensorflow:AutoGraph is not available in this environment: functions lack code information. This is typical of some environments like the interactive Python shell. See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/autograph/g3doc/reference/limitations.md#access-to-source-code for more information.
DIGITE A SENHA123
Digite seu API ID :27372588
Digite seu API HASH :00230fed74fd5954ab090e96c3c78751
start
mini_teste.py:209: RuntimeWarning: coroutine 'idle' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
mini_teste.py:190: RuntimeWarning: coroutine 'idle' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
2023-01-04 18:38:12.299817: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudart64_110.dll'; dlerror: cudart64_110.dll not found
2023-01-04 18:38:12.305757: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cublas64_11.dll'; dlerror: cublas64_11.dll not found
2023-01-04 18:38:12.310141: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cublasLt64_11.dll'; dlerror: cublasLt64_11.dll not found
2023-01-04 18:38:12.316557: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cufft64_10.dll'; dlerror: cufft64_10.dll not found
2023-01-04 18:38:12.322191: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'curand64_10.dll'; dlerror: curand64_10.dll not found
2023-01-04 18:38:12.327431: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cusolver64_11.dll'; dlerror: cusolver64_11.dll not found
2023-01-04 18:38:12.335185: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cusparse64_11.dll'; dlerror: cusparse64_11.dll not found
2023-01-04 18:38:12.340557: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'cudnn64_8.dll'; dlerror: cudnn64_8.dll not found
2023-01-04 18:38:12.345274: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1934] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2023-01-04 18:38:12.359647: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
1
1
1/1 [==============================] - 0s 116ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 80ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
0
0
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 101ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 83ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
WARNING:tensorflow:5 out of the last 5 calls to <function Model.make_predict_function.<locals>.predict_function at 0x000002194A3BA680> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.
1/1 [==============================] - 0s 83ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
56
0
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
WARNING:tensorflow:6 out of the last 6 calls to <function Model.make_predict_function.<locals>.predict_function at 0x000002194A3B8A60> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.
1/1 [==============================] - 0s 122ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 122ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 86ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
99
13
1/1 [==============================] - 0s 209ms/step
1/1 [==============================] - 0s 38ms/step
1/1 [==============================] - 0s 29ms/step
1/1 [==============================] - 0s 28ms/step
1/1 [==============================] - 0s 35ms/step
1/1 [==============================] - 0s 30ms/step
1/1 [==============================] - 0s 20ms/step
1/1 [==============================] - 0s 30ms/step
1/1 [==============================] - 0s 25ms/step
1/1 [==============================] - 0s 110ms/step
1/1 [==============================] - 0s 18ms/step
1/1 [==============================] - 0s 31ms/step
1/1 [==============================] - 0s 29ms/step
Captcha accepted. Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 78ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
79
0
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!
recebeu uma foto
1
1
1/1 [==============================] - 0s 83ms/step
Waiting for captcha ...
Warning! Sometimes captcha image is shown cropped. Tap/click on the image to see the full view!

And doesn´t works. Don´t resolve the captchas. Have some trouble?

**Comment by sethjuarez** [comment]
Hi! Seth here. What are you trying to accomplish? I feel like there may be some context missing.

**Comment by SnowBlueChain** [comment, marked as answer]
It looks like the application is having trouble resolving the captchas. There could be several reasons for this, such as:

The captcha-solving service being used by the application may not be working properly or may be experiencing technical issues.

The captchas being presented to the application may be too difficult to solve, which can happen if they are intentionally designed to be challenging.

There may be connectivity issues between the application and the captcha-solving service, which could prevent the captchas from being properly resolved.

To troubleshoot the issue, you can try the following steps:

Check that the captcha-solving service being used by the application is working properly. You can do this by contacting their support team or checking their website or social media pages for any known issues.

Try using a different captcha-solving service and see if the issue persists. This can help determine if the problem is with the specific service being used or with the application itself.

If the captchas being presented are particularly difficult, consider implementing a different strategy for solving them. For example, you could use machine learning models to automatically identify and solve simple captchas, while more complex captchas could be sent to human workers for manual resolution.

Ensure that there are no connectivity issues between the application and the captcha-solving service. You can do this by checking your internet connection and firewall settings, and ensuring that the necessary ports are open.

**Comment by PriyanshNegi** [comment]
It seems like you're encountering difficulties with resolving captchas using the application. Resolving captchas can be a complex task, and there could be several reasons why it's not working as expected. Here are a few troubleshooting steps you can try:

1. Verify the installation: Make sure you have installed all the necessary dependencies and libraries required by the application. Double-check the installation instructions and ensure that the versions of TensorFlow and other relevant libraries are compatible.

2. Check GPU support: The warning messages you shared indicate that certain GPU libraries are missing. If you are relying on GPU acceleration, ensure that you have the correct GPU drivers and libraries installed. If not, you can try running the application on CPU mode.

3. Review the code: Examine the code you used for resolving captchas and ensure it is implemented correctly. Look for any potential issues, such as missing function calls or improper usage of coroutines.

4. Test with different captchas: Try using different types of captchas to see if the issue is specific to certain formats. Some captchas may require additional preprocessing or specialized algorithms for accurate resolution.

5. Seek community support: If the problem persists, consider reaching out to the application's community or support channels. They might have encountered similar issues and could provide specific guidance or troubleshooting steps.

Remember that resolving captchas can be a challenging task due to the varying complexity and techniques used to prevent automation. It's important to keep up with the latest developments and techniques in captcha solving to ensure the best possible results.

Hope these suggestions help you in troubleshooting the captcha resolution issue. Good luck, and don't hesitate to reach out if you have further questions!
